api/src/main/java/org/cardanofoundation/rosetta/common/services/ProtocolParamServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package org.cardanofoundation.rosetta.common.services;
+
+import javax.annotation.PostConstruct;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.stereotype.Service;
+
+import org.cardanofoundation.rosetta.api.block.model.domain.ProtocolParams;
+import org.cardanofoundation.rosetta.api.block.model.entity.ProtocolParamsEntity;
+import org.cardanofoundation.rosetta.api.block.model.repository.EpochParamRepository;
+import org.cardanofoundation.rosetta.common.mapper.ProtocolParamsToEntity;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ProtocolParamServiceImpl implements ProtocolParamService {
+
+  private final EpochParamRepository epochParamRepository;
+  private final ProtocolParamsToEntity mapperProtocolParams;
+
+  private ProtocolParams cachedProtocolParams;
+
+  @PostConstruct
```

Comment:
This PostConstruct leads to a crash of the intregrationtest. I had the same problem with the genesisblock caching. 
A work-around would be to save not on postconstruct, but when it is queried the first time.

Reply:
Thank you for the information!
I've added some old-style thread-safe code to get parameters on the first call:
https://github.com/cardano-foundation/cardano-rosetta-java/pull/169/files#diff-e1ab139173b73f29bec8b97daa5b2b6d8a04b47a287be793fa90672656284344R25